### PR TITLE
feat: at_file_share demo

### PR DESCRIPTION
**Pull request by murali-shris**
**- What I did**
- created a demo package to test file send and receive through StorJ
**- How I did it**
- file_sender.dart 
    - read a file from local path
    - call at_client encryption service, 
    - upload encrypted file to StorJ 
    - send notification to receiver
- file_receiver.dart 
    - listen to notification
    - get file url and other details from notification
    - down encrypted file from Storj  
    - decrypt encrypted file using at_client encryption service
    - store decrypted file to local path
**- How to verify it**
 - Start local virtual environment
-  follow instructions in README.md

### CODE REVIEW

File: at_file_share/lib/src/service/file_sender.dart [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+      }
+    } on Exception catch (e, trace) {
+      print(trace);
+    } on Error catch (e, trace) {
+      print(trace);
+    }
+  }
+
+  Future<String> _getStorjUrl(String atSign, String fileName) async {
+    var buzzKey = Platform.environment['ATBUZZKEY'];
```
Comment (murali-shris) [inline at_file_share/lib/src/service/file_sender.dart]: @cpswan How can we get the key and nonce url to access getToken API

Comment (cpswan) [reply at_file_share/lib/src/service/file_sender.dart]: The nonce is retrieved from a getNonce service, and the key is a pre-shared key created for that service instance.

File: at_file_share/README.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#TODO
+
+Prerequisite:
+1. Storj account and s3 access/secret keys
```
Comment (gkc) [inline at_file_share/README.md]: s3 ?

File: at_file_share/README.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   1. Login/signup to https://www.storj.io/
+   2. Create new project e.g zetta_demo
```
Comment (gkc) [inline at_file_share/README.md]: Please remove all references to zetta - suggest change everywhere from zetta to acme

File: at_file_share/bin/at_file_share.dart [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+        defaultsTo: 'root.atsign.org',
+        help: 'environment to share the file (dev/prod)');
+}
+
+void printUsage(ArgParser argParser) {
+  print('Usage: dart at_file_share.dart <flags> [arguments]');
+  print(argParser.usage);
+}
+
+void main(List<String> arguments) async {
```
Comment (gkc) [inline at_file_share/bin/at_file_share.dart:32]: Can you change the sender and receiver to use the CLIBase class instead, please? See https://github.com/atsign-foundation/at_demos/blob/trunk/at_rpc_demo/bin/arithmetic_client.dart for example

File: at_file_share/bin/at_file_share.dart [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    }
+    String mode = results['mode'];
+    print('mode: $mode');
+    AtOnboardingPreference atOnboardingPreference = AtOnboardingPreference()
+      ..namespace =
+          'zetta' // unique identifier that can be used to identify data from your app
+      ..atKeysFilePath = results['atKeysFilePath'];
+
+    if (results['env'] == 'dev') {
+      atOnboardingPreference.rootDomain = 'vip.ve.atsign.zone';
```
Comment (gkc) [inline at_file_share/bin/at_file_share.dart]: If you are using CLIBase, it includes an option to set the root domain so this will be unnecessary

**Review by gkc — CHANGES_REQUESTED** [review]

File: at_file_share/README.md (hunk 2 above)
Comment (murali-shris) [reply at_file_share/README.md]: it is access grant with uplink tool. I have made the changes

File: at_file_share/README.md (hunk 3 above)
Comment (murali-shris) [reply at_file_share/README.md]: done

File: at_file_share/bin/at_file_share.dart (hunk 4 above)
Comment (murali-shris) [reply at_file_share/bin/at_file_share.dart:32]: Thanks Gary. done

File: at_file_share/bin/at_file_share.dart (hunk 5 above)
Comment (murali-shris) [reply at_file_share/bin/at_file_share.dart]: Done

**Review by gkc — APPROVED** [review]

**Pull request merged**